plottr/node/histogram.py

```diff
@@ -0,0 +1,224 @@
+  data. User can select over which data axis to perform the histogramming, as
+  well as how many bins to use.
+* :class:`.HistogrammerWidget` -- node widget that allows GUI specification
+  of the user options for the node.
+"""
+
+from typing import Union, Optional, Dict, List, Type
+
+import numpy as np
+from xhistogram.core import histogram
+
+from plottr import QtWidgets
+from ..gui.widgets import FormLayoutWrapper, DimensionCombo
+from ..data.datadict import DataDictBase, MeshgridDataDict
+from .node import Node, NodeWidget, updateOption
+
+
+class _HistogramOptionsWidget(FormLayoutWrapper):
+    """Form widget providing a combo box for histogramming axis selection and an
+    integer spin box for number of bins."""
+    def __init__(self, parent: Optional[QtWidgets.QWidget] = None):
+        super().__init__(
+            parent=parent,
+            elements=[('Hist. axis', DimensionCombo(dimensionType='axes')),
+                      ('# of bins', QtWidgets.QSpinBox())],
+        )
+        self.combo = self.elements['Hist. axis']
+        self.nbins = self.elements['# of bins']
+        self.nbins.setRange(3, 10000)
+
+
+class HistogrammerWidget(NodeWidget):
+    """Node widget for the :class:`.Histogrammer` node."""
+
+    def __init__(self, node: "Histogrammer"):
+        super().__init__(embedWidgetClass=_HistogramOptionsWidget, node=node)
+
+        self.widget: _HistogramOptionsWidget
+        assert self.widget is not None
+        self.widget.combo.connectNode(self.node)
+
+        self.widget.nbins.setValue(node.nbins)
+        self.setAxis(node.histogramAxis)
+
+        self.optSetters = {
+            'histogramAxis': self.setAxis,
+            'nbins': self.widget.nbins.setValue,
+        }
+        self.optGetters = {
+            'histogramAxis': self.getAxis,
+            'nbins': self.widget.nbins.value,
+        }
+
+        self.widget.combo.dimensionSelected.connect(
+            lambda x: self.signalOption('histogramAxis'))
+        self.widget.nbins.editingFinished.connect(
+            lambda: self.signalOption('nbins'))
+
+
+    def getAxis(self) -> Optional[str]:
+        t = self.widget.combo.currentText()
+        if t == 'None':
+            t = None
+        return t
+
+    def setAxis(self, value: Optional[str]) -> None:
+        if value is None:
+            value = 'None'
+        self.widget.combo.setCurrentText(value)
+
+
+class Histogrammer(Node):
+    """A node that replaces the data with a histogram of the data.
+
+    User can select the number of bins that we will use for the histogram, as
+    well as the histogramming axes.
+    Histrogramming will make the dependent dimension an independent, and the
+    new dependent is occurrence. If the dependent data is complex, histogramming
+    adds two new independents, one for the real and one for the imaginary
+    part.
+
+    Properties are:
+
+    :nbins: ``int``
+        number of bins.
+    :reductions: ``Dict[str, (callable, *args, **kwargs)]``
```

Comment:
this property is not present in the class :)

Reply:
oops, looks like i forgot to fully adapt that after copy/past. fixed.